gelf/reader.go

```diff
@@ -64,67 +75,51 @@ func (r *Reader) Read(p []byte) (int, error) {
 func (r *Reader) ReadMessage() (*Message, error) {
 	cBuf := make([]byte, ChunkSize)
 	var (
-		err        error
-		n, length  int
-		cid, ocid  []byte
-		seq, total uint8
-		cHead      []byte
-		cReader    io.Reader
-		chunks     [][]byte
+		err       error
+		n         int
+		chunkHead []byte
+		cReader   io.Reader
 	)
+	var message []byte
 
-	for got := 0; got < 128 && (total == 0 || got < int(total)); got++ {
+	for {
 		if n, err = r.conn.Read(cBuf); err != nil {
```

Comment:
I am a bit unsure how can handle that better, but the read would return any message from any client, so if you wanted to manipulate things you could just send invalid packages for different message ids and start a DDoS.

I think the minimum we should do is keep track of the remote address at the same as the messageID using https://pkg.go.dev/net#UDPConn.ReadFromUDP.

Reply:
yep, it's possible. however, as we discussed earlier, it's out of the scope of this fix